internal/hashing/writer.go

```diff
@@ -28,5 +28,8 @@ func (h *Writer) Write(p []byte) (int, error) {
 
 // Sum returns the hash of all data written so far.
 func (h *Writer) Sum(d []byte) []byte {
+	if h == nil {
```

Comment:
This feels wrong to me. I can see that it's used in `packer_manager.go` so that when a backend does return `nil` for the hash, the `hashing.Writer` is set to nil. I'd be more comfortable to handle this special case in the `PackerManager` instead of here, what do you think?

Reply:
I've moved the check nil check to the packer_manager and its test.